src/buffer/out/AttrRowIterator.cpp

```diff
@@ -1,162 +1,136 @@
-{
-    _increment(1);
-    return *this;
-}
-
-AttrRowIterator AttrRowIterator::operator++(int) noexcept
-{
-    auto copy = *this;
-    _increment(1);
-    return copy;
-}
-
-AttrRowIterator& AttrRowIterator::operator+=(const ptrdiff_t& movement)
-{
-    if (!_exceeded)
-    {
-        if (movement >= 0)
-        {
-            _increment(gsl::narrow<size_t>(movement));
-        }
-        else
-        {
-            _decrement(gsl::narrow<size_t>(-movement));
-        }
-    }
-
-    return *this;
-}
-
-AttrRowIterator& AttrRowIterator::operator-=(const ptrdiff_t& movement)
-{
-    return this->operator+=(-movement);
-}
-
-AttrRowIterator& AttrRowIterator::operator--() noexcept
-{
-    _decrement(1);
-    return *this;
-}
-
-AttrRowIterator AttrRowIterator::operator--(int) noexcept
-{
-    auto copy = *this;
-    _decrement(1);
-    return copy;
-}
-
-const TextAttribute* AttrRowIterator::operator->() const
-{
-    THROW_HR_IF(E_BOUNDS, _exceeded);
-    return &_run->GetAttributes();
-}
-
-const TextAttribute& AttrRowIterator::operator*() const
-{
-    THROW_HR_IF(E_BOUNDS, _exceeded);
-    return _run->GetAttributes();
-}
-
-// Routine Description:
-// - increments the index the iterator points to
-// Arguments:
-// - count - the amount to increment by
-void AttrRowIterator::_increment(size_t count) noexcept
-{
-    while (count > 0)
-    {
-        const size_t runLength = _run->GetLength();
-        if (count + _currentAttributeIndex < runLength)
-        {
-            _currentAttributeIndex += count;
-            return;
-        }
-        else
-        {
-            count -= runLength - _currentAttributeIndex;
-            ++_run;
-            _currentAttributeIndex = 0;
-        }
-    }
-}
-
-// Routine Description:
-// - decrements the index the iterator points to
-// Arguments:
-// - count - the amount to decrement by
-void AttrRowIterator::_decrement(size_t count) noexcept
-{
-    while (count > 0)
-    {
-        // If there's still space within this color attribute to move left, do so.
-        if (count <= _currentAttributeIndex)
-        {
-            _currentAttributeIndex -= count;
-            return;
-        }
-        // If there's not space, move to the previous attribute run
-        // We'll walk through above on the if branch to move left further (if necessary)
-        else
-        {
-            // make sure we don't go out of bounds
-            if (_run == _pAttrRow->_list.cbegin())
-            {
-                _exceeded = true;
-                return;
-            }
-            count -= _currentAttributeIndex + 1;
-            --_run;
-            _currentAttributeIndex = _run->GetLength() - 1;
-        }
-    }
-}
-
-// Routine Description:
-// - sets fields on the iterator to describe the end() state of the ATTR_ROW
-void AttrRowIterator::_setToEnd() noexcept
-{
-    _run = _pAttrRow->_list.cend();
-    _currentAttributeIndex = 0;
-}
+// Copyright (c) Microsoft Corporation.
```

Comment:
A number of these files got CRLF reformatted. Our repo unfortunately stores _CRLF_, not LF, because of how we migrate code back into Windows.

Reply:
(Yes, this is an immense pain.)

Reply:
I just had VS save the files - is there something I should do in my git config to get the right line endings?

Reply:
Hmm. In my experience, it's proven difficult to get git to behave in a Windowsy way.

This may help.

* `git config --local core.autocrlf false`
* `git config --local core.eol CRLF`
* save the file in VS as CRLF
* `git add` it
* `git diff --cached` to ensure the entire file changed
* commit from there

Given that git tries _really hard_ to fix line endings for you, it still might pretend nothing has changed.

Reply:
Personally I’d use VSCode to change line endings of exiting files. I have autocrlf set to off globally to stop git from doing anything “smart”.

Reply:
Unfortunately that didn't work @DHowett - I'm not sure how to correct these line ending shenanigans now 😢  Unless I missed a step in "save the file in VS as CRLF"?  I'm just using File -> Save...

@skyline75489 - I tried opening the files in VS Code but I don't know how to use it to save with different line endings.  Any tips?

It'd be wonderful if 'runformat' could somehow also fix up line endings since that's required to be run anyway.

Reply:
Just to get this one out of the way -- I'm happy to tinker around with your branch and fix the spacing before merge. This is an OpenConsole problem, so an openconsole person should bear the pain of fixing it.

If you've got other updates to push, that might be a good thing to nail down before I go bothering around in your code. ;P

Reply:
(My strategy is going to be, roughly, "try to use git through WSL to make it act right, because clearly it has some additional behavior on Windows that is undesirable" (and then I'll document what I did))

Reply:
@Austin-Lamb At the right bottom section in VSCode:

![图片](https://user-images.githubusercontent.com/4710575/102299023-33630c80-3f8d-11eb-8277-f6aa086a0171.png)

Click the `LF`, VSCode will prompt you about what the line ending would be. 

(Forgive me for the readability of the theme)

Reply:
@DHowett -  I appreciate the help.  I pushed the only other change I was gonna do, so if you can help with line endings that would be great.  I'd love to get this change in before holiday vacations.